csharp/test/Ice/objects/AllTests.cs

```diff
@@ -38,7 +38,7 @@ public bool called
                 }
             }
 
-            public partial struct S1
+            public partial record struct S1
```

Comment:
You'll notice this is the only change to the C# code, which shows a high-level of backward compatibility.